### PR TITLE
212: speaker diarization

### PR DESCRIPTION
**Pull request by eaidova**
TO DO:

- [x] privide binder link
- [x] hide requirements installing output

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -130,6 +130,7 @@ More amazing notebooks here!
...
 | [211-speech-to-text](notebooks/211-speech-to-text/)<br>[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?filepath=notebooks%2F211-speech-to-text%2F211-speech-to-text.ipynb) | Run inference on speech-to-text recognition model | <img src="https://user-images.githubusercontent.com/36741649/140987347-279de058-55d7-4772-b013-0f2b12deaa61.png" width=225>|
+| [212-pyannote-speaker-diarization](notebooks/212-pyannote-speaker-diarization/)<br>[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?filepath=notebooks%2F212-pyannote-speaker-diarization%2F212-pyannote-speaker-diarization.ipynb) | Run inference on spekaer diarization pipeline | <img src="https://user-images.githubusercontent.com/29454499/218432101-0bd0c424-e1d8-46af-ba1d-ee29ed6d1229.png" width=225>|
```
Comment (adrianboguszewski) [inline README.md:133]: I think something similar to "Identify/detect speakers on the recording" would be a better description

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]
This is really nice notebook and I like the diagrams and explanations a lot. My feedback:
- I see some typos, so I recommend using Grammarly for all files once again
- Is there any better audio sample? I can barely hear any person, so I don't know if the model identified all speakers correctly
- OV inference is slower for me :/
image
image

**Comment by eaidova** [comment]
<p>oh...  probably the only way to fix, create fork of repo and update file reading. I consider this as I know one more potential place for error</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/827/discussion/#comment-1428721980'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
@sbalandi problem with installing resolved, please try again

**Comment by eaidova** [comment]
> I see some typos, so I recommend using Grammarly for all files once again
fixed
> Is there any better audio sample? I can barely hear any person, so I don't know if the model identified all speakers correctly
changed
> OV inference is slower for me :/
updated way for running inference

File: README.md (hunk 1 above)
Comment (eaidova) [reply README.md:133]: not sure, speaker verification/identification is slightly different task, when you try to understand who exactly speak on audio, where only one person speak. it does not involves identification when it was saying

File: notebooks/212-pyannote-speaker-diarization/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Spekaer Diarization with OpenVINO™
```
Comment (adrianboguszewski) [inline notebooks/212-pyannote-speaker-diarization/README.md:1]: typo

File: notebooks/README.md [hunk 3]
```diff
@@ -130,6 +130,7 @@ More amazing notebooks here!
...
 | [211-speech-to-text](211-speech-to-text/)<br>[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?filepath=notebooks%2F211-speech-to-text%2F211-speech-to-text.ipynb) | Run inference on speech-to-text recognition model | <img src="https://user-images.githubusercontent.com/36741649/140987347-279de058-55d7-4772-b013-0f2b12deaa61.png" width=225>|
+| [212-pyannote-speaker-diarization](212-pyannote-speaker-diarization/)<br>[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?filepath=notebooks%2F212-pyannote-speaker-diarization%2F212-pyannote-speaker-diarization.ipynb) | Run inference on spekaer diarization pipeline | <img src="https://user-images.githubusercontent.com/29454499/218432101-0bd0c424-e1d8-46af-ba1d-ee29ed6d1229.png" width=225>|
```
Comment (adrianboguszewski) [inline notebooks/README.md:133]: typo

File: README.md (hunk 1 above)
Comment (adrianboguszewski) [reply README.md:133]: I see, but there is still typo

**Review by adrianboguszewski — COMMENTED** [review]
I like the changes. The only issue I can see now is 3 identified speakers instead of 2, but this is related to the model itself, not OV.

**Comment by sbalandi** [comment]
> @sbalandi problem with installing resolved, please try again
Yes, I can download it now, great !!

But I didn't go very far ( Maybe it because I check it on Windows ?
Firstly I stopped on next step (`pipeline = Pipeline.from_pretrained("philschmid/pyannote-speaker-diarization-endpoint")`) and got:
`OSError: [WinError 1314] A required privilege is not held by the client: 'C:\\Users\\sbalandi\\.cache\\huggingface\\hub\\models--speechbrain--spkrec-ecapa-voxceleb\\snapshots\\5c0be3875fda05e81f3c004ed8c7c06be308de1e\\hyperparams.yaml' -> 'C:\\Users\\sbalandi\\.cache\\torch\\pyannote\\speechbrain\\hyperparams.yaml'`

With Administrator rights it can pass, but after I got: 
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[2], line 20
      2 # ## login to huggingfacehub to get access to pretrained model 
      3 # from huggingface_hub import notebook_login, whoami
      4 
   (...)
     15 
     16 # !pip install --upgrade huggingface_hub
     18 from pyannote.audio import Pipeline
---> 20 pipeline = Pipeline.from_pretrained("philschmid/pyannote-speaker-diarization-endpoint")

File c:\users\sbalandi\me\work\1\openvino_env\lib\site-packages\pyannote\audio\core\pipeline.py:134, in Pipeline.from_pretrained(cls, checkpoint_path, hparams_file, use_auth_token, cache_dir)
    131     pipeline.freeze(params)
    133 if "params" in config:
--> 134     pipeline.instantiate(config["params"])
    136 if hparams_file is not None:
    137     pipeline.load_params(hparams_file)

File c:\users\sbalandi\me\work\1\openvino_env\lib\site-packages\pyannote\pipeline\pipeline.py:444, in Pipeline.instantiate(self, params)
    441         continue
    443     msg = f"parameter '{name}' does not exist"
--> 444     raise ValueError(msg)
    446 self.initialize()
    448 return self

ValueError: parameter 'segmentation_onset' does not exist
```

**Review by sbalandi — APPROVED** [review]

**Comment by eaidova** [comment]
> > @sbalandi problem with installing resolved, please try again
> 
> Yes, I can download it now, great !!
> 
> One thing, on Windows `pipeline = Pipeline.from_pretrained("philschmid/pyannote-speaker-diarization-endpoint")` work only with Administrator rights, without it I got: `OSError: [WinError 1314] A required privilege is not held by the client: 'C:\\Users\\sbalandi\\.cache\\huggingface\\hub\\models--speechbrain--spkrec-ecapa-voxceleb\\snapshots\\5c0be3875fda05e81f3c004ed8c7c06be308de1e\\hyperparams.yaml' -> 'C:\\Users\\sbalandi\\.cache\\torch\\pyannote\\speechbrain\\hyperparams.yaml'`

@sbalandi  it is very strange, I can not reproduce reported issue, maybe something broken in caching (e.g. due to unstable connection  during loading)? please try to remove `C:\\Users\\sbalandi\\.cache\\huggingface\\hub\\models--speechbrain--spkrec-ecapa-voxceleb\\` and rerun

File: notebooks/README.md (hunk 3 above)
Comment (eaidova) [reply notebooks/README.md]: fixed, thanks

**Comment by sbalandi** [comment]
> > > @sbalandi problem with installing resolved, please try again
> > 
> > 
> > Yes, I can download it now, great !!
> > One thing, on Windows `pipeline = Pipeline.from_pretrained("philschmid/pyannote-speaker-diarization-endpoint")` work only with Administrator rights, without it I got: `OSError: [WinError 1314] A required privilege is not held by the client: 'C:\\Users\\sbalandi\\.cache\\huggingface\\hub\\models--speechbrain--spkrec-ecapa-voxceleb\\snapshots\\5c0be3875fda05e81f3c004ed8c7c06be308de1e\\hyperparams.yaml' -> 'C:\\Users\\sbalandi\\.cache\\torch\\pyannote\\speechbrain\\hyperparams.yaml'`
> 
> @sbalandi it is very strange, I can not reproduce reported issue, maybe something broken in caching (e.g. due to unstable connection during loading)? please try to remove `C:\\Users\\sbalandi\\.cache\\huggingface\\hub\\models--speechbrain--spkrec-ecapa-voxceleb\\` and rerun

I tried remove and run one more time, the same result, file looks normal and for modifying both folders manually administrative rights are not needed

**Pull request merged**